### PR TITLE
Fix `Home/End` key when header/summary row is selected and validate `rowIdx` in before calling `onCellKeyDown`

### PR DESCRIPTION
**Pull request by amanmahajan7**

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -1019,7 +1017,7 @@ function DataGrid<R, SR, K extends Key>(
         <RowSelectionChangeProvider value={selectRowLatest}>
           <RowSelectionProvider value={allRowsSelected}>
             <HeaderRow
-              columns={getRowViewportColumns(-1)}
+              columns={getRowViewportColumns(minRowIdx)}
```
Comment (amanmahajan7) [inline src/DataGrid.tsx:1020]: We missed it when we added `topSummaryRows`

File: src/TreeDataGrid.tsx [hunk 2]
```diff
@@ -292,8 +292,26 @@ function TreeDataGrid<R, SR, K extends Key>(
     // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
     const idx = column?.idx ?? -1;
     const row = rows[rowIdx];
-    if (!isGroupRow(row)) return;
```
Comment (amanmahajan7) [inline src/TreeDataGrid.tsx:295]: `row` is actually undefined on the header and summary row. I will create a follow-up PR to fix types
https://github.com/adazzle/react-data-grid/blob/main/src/DataGrid.tsx#L515

Comment (amanmahajan7) [reply src/TreeDataGrid.tsx:295]: I will also add more tests in a follow up PR

File: src/TreeDataGrid.tsx [hunk 3]
```diff
@@ -305,24 +323,12 @@ function TreeDataGrid<R, SR, K extends Key>(
         toggleGroup(row.id);
```
Comment (amanmahajan7) [inline src/TreeDataGrid.tsx:323]:
```suggestion
        toggleGroup(row.id);
        return;
```

**Review by nstepien — APPROVED** [review]

**Pull request merged**